src/gurt/telemetry.c

```diff
@@ -541,45 +641,61 @@ d_tm_print_gauge(uint64_t val, struct d_tm_stats_t *stats, char *name,
  * \param[in]	level		Indicates level of indentation when printing
  *				this \a node
  * \param[in]	stream		Direct output to this stream (stdout, stderr)
+ * \param[in]	path		The full path of the node.
+ *				This path is not stored with the node itself.
+ *				This string is passed in so that it may be
+ *				printed for this and children nodes.
+ * \param[in]	show_meta	Set to true to print the meta-data.
+ * \param[in]	format		Output format.
+ *				Choose D_TM_VERBOSE for standard output.
+ *				Choose D_TM_CSV for comma separated values.
  */
 void
 d_tm_print_node(uint64_t *shmem_root, struct d_tm_node_t *node, int level,
-		FILE *stream)
+		FILE *stream, char *path, bool show_meta, int format)
 {
 	struct d_tm_stats_t	stats = {0};
 	struct timespec		tms;
 	uint64_t		val;
 	time_t			clk;
 	char			*name = NULL;
+	char			*shortDesc = NULL;
+	char			*longDesc = NULL;
+	bool			stats_printed = false;
 	int			i = 0;
 	int			rc;
 
 	name = d_tm_conv_ptr(shmem_root, node->dtn_name);
 	if (name == NULL)
 		return;
 
-	for (i = 0; i < level; i++)
-		fprintf(stream, "%20s", " ");
+	if (format == D_TM_CSV) {
+		fprintf(stream, "%s/%s,", path, name);
+	} else {
+		for (i = 0; i < level; i++)
+			fprintf(stream, "%20s", " ");
+	}
 
 	switch (node->dtn_type) {
 	case D_TM_DIRECTORY:
-		fprintf(stream, "%-20s\n", name);
+		if (format != D_TM_CSV)
+			fprintf(stream, "%-20s\n", name);
```

Comment:
The d_tm_print_node() currently prints the full path + name of the metric, for each metric, in CSV mode (which contains all of the directory names up until the final metric name) in order to have complete path/name data for each line of the CSV output.  Whereas, in the standard mode, the directory hierarchy is printed once for all metrics beneath it, since that looks better to show a directory tree on screen.  That is, they are arbitrary style differences which looked appropriate.

Reply:
I think that's fine. It may be helpful to write a brief comment to that effect, as I think it's not that easy to quickly understand the reason for behavior differences while reading the code.